shelley/chain-and-ledger/formal-spec/crypto-primitives.tex

```diff
@@ -67,18 +67,21 @@ \section{Cryptographic primitives}
   \emph{Constraints}
   \begin{align*}
     & \forall (sk, vk) \in \KeyPair,~ d \in \Data,~ \sigma \in \Sig \cdot
-    \sign{sk}{d} = \sigma \implies \verify{vk}{d}{\sigma}
+    \sign{sk}{d} = \sigma \implies (vk, d, \sigma) \in \verify
   \end{align*}
   \emph{Notation for serialized and verified data}
   \begin{align*}
-    & \serialised{x} & \text{serialised representation of } x\\
+    & \serialised{x} & \text{serialised representation of } x (NSF: restrict the type of x?)\\
```

Comment:
I think it's ok as is? Restrict it to what?

Reply:
Hmm. Maybe I meant "indicate" instead of "restrict".

Is `x` allowed to be any data at all? Or maybe just `\Data`? I suppose it's implicitly "any serializable" data.

Reply:
yes,  implicitly "any serializable" data